website/addons/dataverse/static/dataverseFangornConfig.js

```diff
@@ -29,29 +29,55 @@ var _dataverseItemButtons = {
             tb.dropzone.hiddenFileInput.click();
             tb.dropzoneItemCache = item;
         }
-        function dataversePublish(event, item, col) {
+        // Get the external Dataverse host before publishing
+        function getDataverseHost(event, item, col) {
+            var host;
+            $.ajax({
```

Comment:
Instead of sending a request to get the Dataverse host, you can pass in that information from the view so that it's available under `item.data` (`item.data.dataverseIsPublished` is a good example of this)